src/http/controllers/UsersController.ts

```diff
@@ -0,0 +1,152 @@
+  DEFAULT_ERROR: 418,
+};
+
+const apiKey = "707fb022eab54eb1a6434dd36f3eb91c";
+
+class UsersController {
+
+  async create(request: Request, response: Response) {
+
+    try {
+      const { name, email, address, coordinates } = request.body;
+
+      // Verifica se foram fornecidos ambos ou nenhum endereço/coordenadas
+      if ((!address && !coordinates) || (address && coordinates))
+        return response.status(STATUS.BAD_REQUEST).json({ message: 'Provide only address or coordinates, not both or neither.' });
+
+      const geoLocationService = new GeoLocationService(apiKey);
+      const resolvedLocation = await geoLocationService.resolveLocation({ address, coordinates });
+
+      const user = new UserModel({
+        name,
+        email,
+        ...(resolvedLocation.address && { address: resolvedLocation.address }),
+        ...(resolvedLocation.coordinates && { coordinates: resolvedLocation.coordinates }),
+      });
+
+      await user.save();
+
+      return response.status(STATUS.CREATED).json({ message: 'User created successfully', user });
+
+    } catch (error) {
+      //console.error('Error creating user:', error);
+      return response.status(STATUS.INTERNAL_SERVER_ERROR).json({ message: 'Unexpected error' });
+    }
+  }
+
+  async show(request: Request, response: Response): Promise<Response> {
+    //route GET users/show
+
+    try {
+      const { page , limit} = request.query;
+
+      const [users, total] = await Promise.all([
+        UserModel.find().lean(),
+        UserModel.count(),
+      ]);
+
+      if (total === undefined) {
+        return response.status(STATUS.DEFAULT_ERROR).json({ message: 'Failed to count users' });
+      }
+
+      return response.status(STATUS.OK).json({
+        rows: users,
+        page,
+        limit,
+        total,
+      });
+
+    } catch (error) {
+      //console.error('Error when searching for users:', error);
+      return response.status(STATUS.DEFAULT_ERROR).json({ message: 'Unexpected error' });
+    }
+  }
+
+  async findOne(request: Request, response: Response) {
+    // route GET users/user/:id
+
+    try {
+      const { id } = request.params;
+
+      const user = await UserModel.findOne({ _id: id }).lean();
+
+      if (!user) {
+        return response.status(STATUS.INTERNAL_SERVER_ERROR).json({ message: 'User not found' });
+      }
+
+      return response.status(STATUS.OK).json(user);
+
+    } catch (error) {
+      //console.error('Error when searching for user :', error);
+      return response.status(STATUS.DEFAULT_ERROR).json({ message: 'Unexpected error' });
+    }
+  }
+
+  async update(request: Request, response: Response) {
+    try {
+      const userId = request.params.id;
+      const { name, email, address, coordinates } = request.body;
+
+      // Verifica se o ID do usuário está presente nos parâmetros da solicitação
+      if (!userId)
+        return response.status(STATUS.BAD_REQUEST).json({ message: 'User ID is required for updating.' });
```

Comment:
esse check não faz sentido, a rota só vai ser chamada se esse parametro for passado